src/systemctl/fuzz-systemctl-parse-argv.c

```diff
@@ -0,0 +1,51 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include <stdio.h>
+#include <unistd.h>
+
+#include "env-util.h"
+#include "fs-util.h"
+#include "fuzz.h"
+#include "strv.h"
+#include "systemctl.h"
+
+int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
+        _cleanup_strv_free_ char **argv = NULL;
+        _cleanup_free_ char *orig_stdout = NULL;
+        int r;
+
+        /* We don't want to fill the logs with messages about parse errors.
+         * Disable most logging if not running standalone */
+        if (!getenv("SYSTEMD_LOG_LEVEL"))
+                log_set_max_level(LOG_CRIT);
+
+        arg_pager_flags = PAGER_DISABLE; /* We shouldn't execute the pager */
+
+        argv = strv_parse_nulstr((const char *)data, size);
```

Comment:
From experience in other projects, the input data can be huge and reworking it might very well mean OOM quite often - what happens if the fuzzer function returns -ENOMEM? Do we get a ticket on oss-fuzz or is it just ignored?

Reply:
I think the return value is ignored, as mentioned in https://github.com/systemd/systemd/pull/18504.

Reply:
Ah that's good to know, thanks

Reply:
Also, unless configured otherwise, there's a limit of 4096 bytes on the input size.